contracts/PriceCalculator.sol

```diff
@@ -0,0 +1,160 @@
+     * which in turn equals to (100 + a) ** log2(x) / 100 ** log2(x).
+     * Therefore numerator = (100 + a) ** log2(x) and denominator = 100 ** log2(x).
+     * @param _growthFactor         The % growth in price per factor-increase (aka 'c', between 1 and 100).
+     * @param _currTokenIdx         The current token index.
+     */
+    function _subFormulaPower(uint256 _growthFactor, uint256 _currTokenIdx) private pure returns (uint256) {
+        // function _subFormulaPower(uint256 _growthFactor, uint256 _currTokenIdx) private pure returns (uint256) {
+        uint256 exponent = logarithm2(_currTokenIdx);
+        uint256 numeratorLog = power(100 + _growthFactor, exponent);
+        uint256 denominatorLog = power(100, exponent);
+        return numeratorLog.div(denominatorLog);
+    }
+
+    /**
+     * @dev For power operations within unchecked blocks to prevent integer overflow wrapping.
+     */
+    function power(uint256 base, uint256 exponent) private pure returns (uint256 result) {
+        result = base**exponent;
+    }
+
+    /**
+     * @dev Calculates the logarithmic expression with 2 as the logarithmic base, rounds up answer to the nearest integer.
+     *      Taken from https://ethereum.stackexchange.com/questions/8086/logarithm-math-operation-in-solidity
+     *      <700 gas
+     * @param argument Where 2 ** result = argument.
+     */
+    function logarithm2(uint256 argument) internal pure returns (uint256 result) {
+        assembly {
+            let arg := argument
+            argument := sub(argument, 1)
+            argument := or(argument, div(argument, 0x02))
+            argument := or(argument, div(argument, 0x04))
+            argument := or(argument, div(argument, 0x10))
+            argument := or(argument, div(argument, 0x100))
+            argument := or(argument, div(argument, 0x10000))
+            argument := or(argument, div(argument, 0x100000000))
+            argument := or(argument, div(argument, 0x10000000000000000))
+            argument := or(argument, div(argument, 0x100000000000000000000000000000000))
+            argument := add(argument, 1)
+            let m := mload(0x40)
+            mstore(m, 0xf8f9cbfae6cc78fbefe7cdc3a1793dfcf4f0e8bbd8cec470b6a28a7a5a3e1efd)
+            mstore(add(m, 0x20), 0xf5ecf1b3e9debc68e1d9cfabc5997135bfb7a7a3938b7b606b5b4b3f2f1f0ffe)
+            mstore(add(m, 0x40), 0xf6e4ed9ff2d6b458eadcdf97bd91692de2d4da8fd2d0ac50c6ae9a8272523616)
+            mstore(add(m, 0x60), 0xc8c0b887b0a8a4489c948c7f847c6125746c645c544c444038302820181008ff)
+            mstore(add(m, 0x80), 0xf7cae577eec2a03cf3bad76fb589591debb2dd67e0aa9834bea6925f6a4a2e0e)
+            mstore(add(m, 0xa0), 0xe39ed557db96902cd38ed14fad815115c786af479b7e83247363534337271707)
+            mstore(add(m, 0xc0), 0xc976c13bb96e881cb166a933a55e490d9d56952b8d4e801485467d2362422606)
+            mstore(add(m, 0xe0), 0x753a6d1b65325d0c552a4d1345224105391a310b29122104190a110309020100)
```

Comment:
I need a little more time to digest what the code is really doing here.
As someone else was asking [`please explain!`](https://ethereum.stackexchange.com/questions/8086/logarithm-math-operation-in-solidity).